flutter/shell/platform/tizen/tizen_autofill.cc

```diff
@@ -0,0 +1,183 @@
+        autofill_fill_response_foreach_group(
+            fill_response,
+            [](autofill_fill_response_group_h group_h, void* user_data) {
+              autofill_fill_response_group_foreach_item(
+                  group_h,
+                  [](autofill_fill_response_item_h item, void* user_data) {
+                    char* id = nullptr;
+                    char* value = nullptr;
+                    char* presentation_text = nullptr;
+
+                    autofill_fill_response_item_get_id(item, &id);
+                    autofill_fill_response_item_get_presentation_text(
+                        item, &presentation_text);
+                    autofill_fill_response_item_get_value(item, &value);
+
+                    std::unique_ptr<AutofillItem> response_item =
+                        std::make_unique<AutofillItem>();
+                    response_item->label_ = std::string(presentation_text);
+                    response_item->id_ = std::string(id);
+                    response_item->value_ = std::string(value);
+
+                    TizenAutofill::GetInstance().StoreResponseItem(
+                        move(response_item));
+
+                    if (id) {
+                      free(id);
+                    }
+
+                    if (value) {
+                      free(value);
+                    }
+
+                    if (presentation_text) {
+                      free(presentation_text);
+                    }
+
+                    return true;
+                  },
+                  group_h);
+              return true;
+            },
+            &count);
+        TizenAutofill::GetInstance().CallPopupCallback();
+      },
+      nullptr);
+
+  response_items_.clear();
+}
+
+void TizenAutofill::RequestAutofill(std::vector<std::string> hints,
+                                    std::string id) {
+  char* app_id = nullptr;
+  app_get_id(&app_id);
+
+  autofill_view_info_h view_info = nullptr;
+  autofill_view_info_create(&view_info);
+  autofill_view_info_set_app_id(view_info, app_id);
+  autofill_view_info_set_view_id(view_info, id.c_str());
+
+  for (auto hint : hints) {
+    std::optional<autofill_hint_e> autofill_hint = ConvertAutofillHint(hint);
+    if (autofill_hint.has_value()) {
+      autofill_item_h item;
+      autofill_item_create(&item);
+      autofill_item_set_autofill_hint(item, autofill_hint.value());
+      autofill_item_set_id(item, id.c_str());
+      autofill_item_set_sensitive_data(item, false);
+      autofill_view_info_add_item(view_info, item);
+      autofill_item_destroy(item);
+    }
+  }
+
+  int ret = autofill_fill_request(autofill_, view_info);
+  if (ret != AUTOFILL_ERROR_NONE) {
+    FT_LOG(Error) << "autofill_fill_request error";
+  }
+  autofill_view_info_destroy(view_info);
+
+  response_items_.clear();
+}
+
+void TizenAutofill::RegisterAutofillItem(std::string view_id,
+                                         AutofillItem item) {
+  autofill_save_item_h si_h;
```

Comment:
How about `autofill_save_item` or `save_item` instead `si_h`?

Reply:
Here's the guideline.

https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

> Minimize the use of abbreviations that would likely be unknown to someone outside your project (especially acronyms and initialisms).

Also make sure to initialize handles to `nullptr` whenever possible.